File: 백준/Bronze/10798. 세로읽기/세로읽기.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");
let max = 0;
for(let i =0; i<input.length; i++){
  if(max < Number(input[i].length)) max = Number(input[i].length);
}

let result = new Array(max).fill("");
for(let i =0; i<input.length; i++){
  let arr = input[i].split("");
  for(let j  = 0; j<arr.length; j++){
    result[j] = result[j] + arr[j];
  }
}
console.log(result.join(""));

File: 백준/Bronze/10798. 세로읽기/세로읽기.py
import sys
input = sys.stdin.readline

arr = [list(input().strip()) for _ in range(5)]
max_l = 0

for a in arr:
    max_l = max(len(a) , max_l)
    

result = ""

for i in range(max_l):
    for j in range(5):
        if i >= len(arr[j]):
            continue
        else:
            result += arr[j][i]
            
print(result)
